### PR TITLE
[BugFix][Model] Fix Kimi K3 dense MLP sequence parallelism

### PR DESCRIPTION
**Pull request by q664171689**
### What this PR does / why we need it?

Fixes the dense Kimi K3 MLP path when sequence parallelism is enabled.

The dense MLP now gathers sequence-sharded token rows before executing the TP-sharded weights, disables the upstream output reduction for this path, and reduce-scatters the output afterwards. This keeps every TP shard aligned on the same tokens and avoids mixing live and padding rows during reduction.

### Does this PR introduce any user-facing change?

Yes. Kimi K3 workloads that enable sequence parallelism now use the corrected dense-MLP collective sequence.

### How was this patch tested?

- Added `test_kimi_dense_mlp_gathers_and_scatters_sequence_shards`, which verifies the gather → MLP → reduce-scatter sequence and the gathered/sharded tensors using CPU mocks.
- Ran Python syntax compilation for the changed implementation and test files.
- Ran `git diff --check` and `git show --check`.
- The local machine does not have the main branch's vLLM/Ascend test runtime or Ascend NPU hardware, so the targeted pytest and NPU validation are left for project CI.
- vLLM main: https://github.com/vllm-project/vllm/commit/a97dacb7106ee49f39f3d1fc6ae1800ff724e01d

### CODE REVIEW

**Review by Ronald1995 — APPROVED** [review]

File: vllm_ascend/models/kimi_k3.py [hunk 1]
```diff
@@ -125,6 +125,44 @@ def _apply_ascend_attn_res(
...
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.use_sequence_parallel:
+            # All weight shards must operate on the same tokens. Reducing
+            # different sequence shards would mix live and padding rows.
+            x = sp_all_gather(x)
+        x = super().forward(x)
+        if self.use_sequence_parallel:
+            x = sp_reduce_scatter(x)
+        return x
```
Comment (Ronald1995) [inline vllm_ascend/models/kimi_k3.py:163]: please check if vllm support this operation.

Comment (q664171689) [reply vllm_ascend/models/kimi_k3.py:163]:
```python
class KimiMLP(nn.Module):
    def __init__(
        self,
        hidden_size: int,
        intermediate_size: int,
        hidden_act: str,
        quant_config: QuantizationConfig | None = None,
        reduce_results: bool = True,
        prefix: str = "",
        activation_situ_beta: float | None = None,
        activation_situ_linear_beta: float | None = None,
    ) -> None:
        super().__init__()

        self.gate_up_proj = MergedColumnParallelLinear(
            hidden_size,
            [intermediate_size] * 2,
            bias=False,
            quant_config=quant_config,
            prefix=f"{prefix}.gate_up_proj",
        )
        self.down_proj = RowParallelLinear(
            intermediate_size,
            hidden_size,
            bias=False,
            quant_config=quant_config,
            reduce_results=reduce_results,
            prefix=f"{prefix}.down_proj",
        )
        if hidden_act == "silu":
            self.act_fn = SiluAndMul()
        elif hidden_act == "situ":
            self.act_fn = SituAndMul(
                beta=activation_situ_beta or 1.0,
                linear_beta=activation_situ_linear_beta,
            )
        else:
            raise ValueError(
                f"Unsupported activation: {hidden_act}. "
                "Only silu and situ are supported."
            )

    def forward(self, x):
        gate_up, _ = self.gate_up_proj(x)
        x = self.act_fn(gate_up)
        x, _ = self.down_proj(x)
        return x
```
The upstream does not yet support this operation.

**Review by linfeng-yuan — APPROVED** [review]
